.github/workflows/stale-repos.yml

```diff
@@ -1,19 +1,35 @@
 on:
   schedule:
-  - cron: "0 */6 * * *" # every 6 hours
+    - cron: "0 */6 * * *" # every 6 hours
+  pull_request:
+    paths:
+      - 'configs/*.json'
+
+name: Check Stale Repos
 
 jobs:
   matrix:
     name: Find stale repositories
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v2
+      - id: repos
+        name: find repositories to check
+        run: |
+          repos="$(jq '{"\(input_filename)":(.repositories | map(.target))}' configs/*.json | jq -sc 'add')"
+          if [[ '${{ github.event_name }}' == 'pull_request' ]]; then
+            git fetch origin ${{ github.event.pull_request.base.sha }}
```

Comment:
Doesn't the checkout step automatically check out the current state of the repo?

Reply:
It checks out a shallow clone (with `--depth=1`). That's why we don't have any info on the PR target commit unless we explicitly fetch it.

Reply:
Right. Now I remember fighting this a while back ;)